modules/integration/src/test/scala/scala/cli/integration/CompileTestDefinitions.scala

```diff
@@ -540,4 +540,38 @@ abstract class CompileTestDefinitions(val scalaVersionOpt: Option[String])
       expect(classFiles.contains(os.rel / "Hello.class"))
     }
   }
+
+  private def compilerArtifactName: String =
+    if (actualScalaVersion.startsWith("3")) "scala3-compiler" else "scala-compiler"
+
+  test(s"ensure the -with-compiler option $compilerArtifactName to the classpath") {
+    TestInputs(os.rel / "s.sc" -> """println("Hello")""")
+      .fromRoot { root =>
+        val compileRes = os.proc(
+          TestUtil.cli,
+          "compile",
+          "s.sc",
+          "-with-compiler",
+          "--print-classpath",
+          extraOptions
+        )
+          .call(cwd = root)
+        expect(compileRes.out.trim().contains(compilerArtifactName))
```

Comment:
condition from the next test could be added to this test, when the `-with-compiler` option is not passed. It should be much faster, because we have already compiled sources.